lib/tasks/trn_generator.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class TRNGenerator
+  class << self
+    def next
+      available.pop || (raise "TRN available list exhausted")
+    end
+
+  private
+
+    ALL_TRNS = ("0011111".."9999999").to_a.freeze unless defined?(ALL_TRNS)
```

Comment:
This is at least 700MB, right?

Reply:
10 milion TRNs, times 7, times however many bytes per character, times however inefficient Ruby is with it. I would hope closer to 70MB than 700, but it clearly makes our app unhappy 😢

Reply:
Indeed. Looks like we were running at somewhere between 60% and 80% utilization

Reply:
I've changed the strategy to keep these as integers and generate a unique map. That should minimise the hit on it.